.github/scripts/validate-native-config-files.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set -e
+
+# Run the generate-native-configs.sh script
+printf "\nRunning generate-native-configs.sh..."
+./generate-native-configs.sh
+
+# Check for git diff
+printf "\nValidating that there are no changes in the repository..."
+if [[ -n $(git diff --ignore-blank-lines) ]]; then
+  printf "\nError: Running generate-native-configs.sh resulted in changes to the repository."
```

Comment:
This is just an idea and can be ignored. More as a food for thoughts.

 1. '\n' at the end of the string for better terminal formatting.
 2. Added '>&2' to redirect this error message to standard error (stderr).
    This is best practice, allowing users to separate normal output (stdout)
    from error/diagnostic messages (stderr). For example:
    ./validate-native-config-files.sh > log.txt (errors still go to console)
    ./validate-native-config-files.sh > log.txt 2> error.txt (separate files)

```suggestion
  printf "\nError: Running generate-native-configs.sh resulted in changes to the repository.\n" >&2
```

Reply:
Why not just use `echo`? That's what I've typically seen in scripts, I've rarely seen `printf`. `echo` lets you forget about the lines.

This script is meant to be run in a workflow, so if we save to logs, we'd have to upload those as artifacts as well. For such a small feature I wouldn't do that – this is just meant to give the user a heads-up that they should take a look at their config files on their own.

Reply:
chatgtp sugessted me to use echo as it's equivalent of println - let me know if there are any conerns, because I don't know how to do it better